core/rawdb/bor_receipts.go

```diff
@@ -37,51 +37,29 @@ func ReadBorReceiptRLP(db kv.Getter, hash common.Hash, number uint64) rlp.RawVal
 	return data
 }
 
-// ReadRawBorReceipt retrieves the block receipt belonging to a block.
-// The receipt metadata fields are not guaranteed to be populated, so they
-// should not be used. Use ReadBorReceipt instead if the metadata is needed.
-func ReadRawBorReceipt(db kv.Tx, hash common.Hash, number uint64) *types.Receipt {
-	// Retrieve the flattened receipt slice
-	data := ReadBorReceiptRLP(db, hash, number)
-	if len(data) == 0 {
-		return nil
-	}
-
-	// Convert the receipts from their storage form to their internal representation
-	var storageReceipt types.ReceiptForStorage
-	if err := rlp.DecodeBytes(data, &storageReceipt); err != nil {
-		log.Error("Invalid receipt array RLP", "hash", hash, "err", err)
-		return nil
-	}
-
-	return (*types.Receipt)(&storageReceipt)
-}
-
 // ReadBorReceipt retrieves all the bor block receipts belonging to a block, including
 // its correspoinding metadata fields. If it is unable to populate these metadata
 // fields then nil is returned.
-func ReadBorReceipt(db kv.Tx, hash common.Hash, number uint64) *types.Receipt {
+func ReadBorReceipt(db kv.Tx, hash common.Hash, number uint64) (*types.Receipt, error) {
 	// We're deriving many fields from the block body, retrieve beside the receipt
-	borReceipt := ReadRawBorReceipt(db, hash, number)
-	if borReceipt == nil {
-		return nil
+	data, err := db.GetOne(kv.BorReceipts, borReceiptKey(number))
+	if err != nil {
+		return nil, err
```

Comment:
add context to this error - so we know that this error came from `ReadBorReceipt` and witch which params.